pep/app/utils/search.ts

```diff
@@ -1,57 +1,111 @@
     sourcetype?: string;
     articletype?: string;
+    sourcecode?: string;
+    sourcename?: string;
     sourcelangcode?: string;
     title?: string;
+    volume?: string;
+    issue?: string;
     startyear?: string;
     endyear?: string;
     citecount?: string;
     viewcount?: string;
+    viewperiod?: string;
 }
 
 interface SearchQueryParams extends SearchQueryStrParams {
     facetfields: string | null;
     synonyms: boolean;
+    abstract: boolean;
+}
+
+export interface SearchFacetCounts {
+    [x: string]: number;
 }
 
+/**
+ * Builds the query params object for document searches
+ * @export
+ * @param {string} smartSearchTerm
+ * @param {SearchTermValue[]} searchTerms
+ * @param {boolean} synonyms
+ * @param {SearchFacetValue[]} [facetValues=[]]
+ * @param {string[]} [facetFields=DEFAULT_SEARCH_FACETS]
+ * @param {('AND' | 'OR')} [logicalOperator='OR']
+ * @returns {QueryParamsObj}
+ */
 export function buildSearchQueryParams(
     smartSearchTerm: string,
     searchTerms: SearchTermValue[],
     synonyms: boolean,
     facetValues: SearchFacetValue[] = [],
-    facetFields: string[] = defaultFacetFields,
-    logicalOperator: 'AND' | 'OR' = 'OR'
-) {
+    citedCount: string = '',
+    viewedCount: string = '',
+    viewedPeriod: ViewPeriod | null = null,
+    facetFields: string[] = DEFAULT_SEARCH_FACETS,
+    joinOp: 'AND' | 'OR' = 'AND'
+): QueryParamsObj {
     const queryParams: SearchQueryParams = {
         facetfields: !isEmpty(facetFields) ? facetFields.join(',') : null,
-        fulltext1: smartSearchTerm,
+        smarttext: smartSearchTerm,
+        citecount: citedCount,
+        viewcount: viewedCount,
+        viewperiod: `${!isNone(viewedPeriod) && !isEmpty(viewedCount) ? viewedPeriod : ''}`,
+        abstract: true,
         synonyms
     };
 
     const nonEmptyTerms = searchTerms.filter((t) => !!t.term);
+    const parascopes: string[] = [];
 
     nonEmptyTerms.forEach((term) => {
         let searchType = SEARCH_TYPES.findBy('id', term.type);
         if (searchType && searchType.param) {
-            //if a term of this type already exists, join it to the existing one
+            let value = term.term.trim();
             const p = searchType.param as keyof SearchQueryStrParams;
-            if (queryParams[p]) {
-                queryParams[p] = `${queryParams[p]} ${logicalOperator} ${term.term}`;
-            } else {
-                queryParams[p] = term.term;
+            const qp = queryParams[p];
+
+            if (searchType.solrField) {
+                const isQuoted = QUOTED_VALUE_REGEX.test(value);
+                const hasMultipleWords = value.indexOf(' ') !== -1;
+                const formatted = hasMultipleWords && !isQuoted ? `"${value}"~25` : value;
+                value = `${searchType.solrField}:(${formatted})`;
+            }
+
+            //if a term of this type already exists, join it to the existing one
+            queryParams[p] = `${qp ? `${qp} ${joinOp} ` : ''}${value}`;
```

Comment:
Seems like you could create a small method for this since its used in multiple places